warp10/src/main/sh/warp10.sh

```diff
@@ -525,133 +391,168 @@ start() {
 }
 
 stop() {
-
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
-
+  isWarp10User
   if isStarted; then
     echo "Stopping Warp 10..."
-    kill $(cat ${PID_FILE})
+    kill "$(cat "${PID_FILE}")"
     echo "Waiting for Warp 10 to stop..."
-    while $(kill -0 $(cat ${PID_FILE}) 2>/dev/null); do
+    while $(kill -0 $(cat "${PID_FILE}") 2>/dev/null); do
       sleep 2
     done
     echo "Warp 10 stopped..."
-    rm -f ${PID_FILE}
+    rm -f "${PID_FILE}"
   else
     echo "No instance of Warp 10 is currently running"
   fi
 }
 
 status() {
-
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
-
   if isStarted; then
-    ps -Ao pid,etime,args | grep "^\s*$(cat ${PID_FILE})\s"
+    ps -Ao pid,etime,args | grep "^\s*$(cat "${PID_FILE}")\s"
   else
     echo "No instance of Warp 10 is currently running"
   fi
-  
+
   checkRam
 }
 
 tokengen() {
   if [ "$#" -ne 2 ]; then
-    echo "Usage: $0 tokengen envelope.mc2"
-    exit 1
+    die "Usage: $0 tokengen envelope.mc2"
   fi
-  getConfigFiles
-  ${JAVACMD} -cp ${WARP10_CP} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 io.warp10.TokenGen ${CONFIG_FILES} $2
+  ${JAVACMD} -cp "${WARP10_JAR}" -Dlog4j.configuration=file:"${LOG4J_CONF}" -Dfile.encoding=UTF-8 io.warp10.TokenGen ${CONFIG_FILES} "$2" 2>/dev/null
 }
 
 run() {
-  getConfigFiles
-  ${JAVACMD} -cp ${WARP10_CP} -Dlog4j.configuration=file:${LOG4J_CONF} -Dfile.encoding=UTF-8 -Dwarp10.config="${CONFIG_FILES}" io.warp10.WarpRun $2
+  ${JAVACMD} -cp "${WARP10_JAR}" -Dlog4j.configuration=file:"${LOG4J_CONF}" -Dfile.encoding=UTF-8 -Dwarp10.config=${CONFIG_FILES} io.warp10.WarpRun "$2"
 }
 
 repair() {
-
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
+  isWarp10User
 
   if isStarted; then
-    echo "Operation has been cancelled! - Warp 10 instance must be stopped for repair"
-    exit 1
+    die "Operation has been cancelled! - Warp 10 instance must be stopped for repair"
   else
     echo "Repairing LevelDB..."
-    ${JAVACMD} -cp ${WARP10_JAR} -Dfile.encoding=UTF-8 io.warp10.standalone.WarpRepair ${LEVELDB_HOME}
+    ${JAVACMD} -cp "${WARP10_JAR}" -Dfile.encoding=UTF-8 io.warp10.standalone.WarpRepair "${LEVELDB_HOME}"
   fi
 }
 
 compact() {
-  #
-  # Make sure the caller is WARP10_USER
-  #
-  isUser ${WARP10_USER}
+  isWarp10User
 
   if isStarted; then
-    echo "Operation has been cancelled! - Warp 10 instance must be stopped for compaction"
-    exit 1
+    die "Operation has been cancelled! - Warp 10 instance must be stopped for compaction"
   else
     echo "Compacting LevelDB..."
-    ${JAVACMD} -cp ${WARP10_JAR} io.warp10.leveldb.WarpCompact ${LEVELDB_HOME} $1 $2
+    ${JAVACMD} -cp "${WARP10_JAR}" io.warp10.leveldb.WarpCompact "${LEVELDB_HOME}" "$1" "$2"
   fi
 }
 
+##
+## Initialize script
+##
+getWarp10Home
+WARP10_REVISION=@VERSION@
```

Comment:
No, warp10.sh is linked to the version jar (because we may change launch / class / jvm options during 3.x lifetime), and should always be updated with the jar. To keep several old version, it is easy to rename warp10.sh to warp10-previosxxx.sh during the manual update process. Or warp10.sh could symlink to ./warp10-3.x.sh (and the symlink included in the tgz).

Reply:
So the version should not be present in warp10-env.sh

Reply:
It isn't (there is no `@VERSION@` inside).

Reply:
then line 58 of package.sh should be removed

Reply:
done